File: resto_caisse_app/lib/models/serveur.dart
class Serveur {
  final int id;
  final String nom;
  final String? prenom;
  final String? telephone;

  Serveur({
    required this.id,
    required this.nom,
    this.prenom,
    this.telephone,
  });

  factory Serveur.fromJson(Map<String, dynamic> json) {
    return Serveur(
      id: json['id'],
      nom: json['nom'],
      prenom: json['prenom'],
      telephone: json['telephone'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nom': nom,
      'prenom': prenom,
      'telephone': telephone,
    };
  }
}

File: app/Models/Serveur.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Serveur extends Model
{
    use HasFactory;

    protected $fillable = [
        'nom',
        'prenom',
        'telephone',
        'actif',
    ];

    protected $casts = [
        'actif' => 'boolean',
    ];

    public function commandes()
    {
        return $this->hasMany(Commande::class);
    }
}
